network/framework/src/application/storage.rs

```diff
@@ -91,6 +103,69 @@ impl PeersAndMetadata {
         all_peers
     }
 
+    /// Return copy of list of peers if generation is different than previously held.
+    /// May optionally filter for connected peers only.
+    /// May optionally filter on match for any of a set of ProtocolId.
+    pub fn get_all_peers_generational(
```

Comment:
Curious: what's the benefit of this over the existing cache? i.e., we currently have a single cache location with cheap reads/writes, but if we move to the generational API, we'll have multiple cached instances (each application will need to maintain a cache of what it last saw and update based on if these APIs return new generations?). Have we seen a bottleneck somewhere to justify moving to the new API?